### PR TITLE
feat(customer): customer v2 refactor for customer create end point

### PR DESCRIPTION
**Pull request by sahkal**
## Type of Change

- [x] Refactoring

## Description

 customer v2 refactor for customer create end point

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Create a Customer and test sanity for v1 endpoints.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/router/src/core/customers.rs [hunk 1]
```diff
@@ -33,122 +41,329 @@ pub async fn create_customer(
...
-                customer_address,
-                merchant_id,
-                &customer_id,
-                key,
-                merchant_account.storage_scheme,
-            )
-            .await
-            .switch()
-            .attach_printable("Failed while encrypting address")?;
+    merchant_ref_id_customer_struct
```
Comment (jarnura) [inline crates/router/src/core/customers.rs:67]:
```suggestion
    merchant_reference_id_customer
```

**Review by jarnura — CHANGES_REQUESTED** [review]

**Review by jarnura — DISMISSED** [review]

**Review by jarnura — DISMISSED** [review]

File: crates/api_models/src/events/customer.rs [hunk 2]
```diff
@@ -5,31 +5,33 @@ use crate::customers::{CustomerDeleteResponse, CustomerId, CustomerRequest, Cust
...
+                customer_id: Some(customer_id),
+            })
     }
 }
 
 impl ApiEventMetric for CustomerResponse {
     fn get_api_event_type(&self) -> Option<ApiEventsType> {
         Some(ApiEventsType::Customer {
-            customer_id: self.customer_id.clone(),
+            customer_id: self.get_merchant_reference_id().clone(),
```
Comment (lsampras) [inline crates/api_models/src/events/customer.rs:26]: For now lets keep the customer_id as mandatory,
in places where we don't have a customer id we can just not log the event for now by returning a `None`

```rust
impl ApiEventMetric for CustomerResponse {
    fn get_api_event_type(&self) -> Option<ApiEventsType> {
        self.get_merchant_reference_id().clone().map(|cid| 
            ApiEventsType::Customer { customer_id: cid } 
        )
    }
}
```

File: crates/common_utils/src/events.rs [hunk 3]
```diff
@@ -28,7 +28,7 @@ pub enum ApiEventsType {
         payment_method_type: Option<PaymentMethodType>,
     },
     Customer {
-        customer_id: id_type::CustomerId,
+        customer_id: Option<id_type::CustomerId>,
```
Comment (lsampras) [inline crates/common_utils/src/events.rs:31]: lets keep this mandatory

**Review by lsampras — CHANGES_REQUESTED** [review]
rest looks good

**Review by lsampras — DISMISSED** [review]

**Review by lsampras — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Pull request merged**